### PR TITLE
Fix gradle BuildConfig logic

### PR DESCRIPTION
**Pull request by sjaindl**
Fixes generation of BuildConfig after Android Studio + gradle update.

Testrun pending:
https://jenkins.catrob.at/view/All-Categories/view/Catroid-multi-job/job/Catroid-Multi-Job-Custom-Branch-RELOADED/2461/

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -26,22 +26,46 @@ apply from: 'gradle/intellij_config_tasks.gradle'
...
+            buildConfigField "boolean", "FEATURE_LED_BRICK_ENABLED", "true"
+            buildConfigField "boolean", "FEATURE_VIBRATION_BRICK_ENABLED", "true"
+            buildConfigField "boolean", "FEATURE_BACKPACK_ENABLED", "true"
+            buildConfigField "boolean", "FEATURE_PARROT_AR_DRONE_ENABLED", "true"
+            buildConfigField "boolean", "FEATURE_APK_GENERATOR_ENABLED", "true"
+            buildConfigField "boolean", "FEATURE_COMPUTER_VISION_ENHANCEMENT_ENABLED", "true"
+            buildConfigField "boolean", "FEATURE_FORMULA_EDITOR_LISTS_ENABLED", "true"
+            buildConfigField "boolean",
+                    "FEATURE_PHYSICS_ENGINE_COLLISSION_FILTERING_ENABLED",
+                    "true"
```
Comment (robertpainsi) [inline build.gradle]: Please don't wrap this line.

File: build.gradle [hunk 2]
```diff
@@ -26,22 +26,46 @@ apply from: 'gradle/intellij_config_tasks.gradle'
...
+            buildConfigField "boolean", "FEATURE_LED_BRICK_ENABLED", "false"
+            buildConfigField "boolean", "FEATURE_VIBRATION_BRICK_ENABLED", "false"
+            buildConfigField "boolean", "FEATURE_BACKPACK_ENABLED", "false"
+            buildConfigField "boolean", "FEATURE_PARROT_AR_DRONE_ENABLED", "false"
+            buildConfigField "boolean", "FEATURE_APK_GENERATOR_ENABLED", "false"
+            buildConfigField "boolean", "FEATURE_COMPUTER_VISION_ENHANCEMENT_ENABLED", "true"
+            buildConfigField "boolean", "FEATURE_FORMULA_EDITOR_LISTS_ENABLED", "true"
+            buildConfigField "boolean",
+                    "FEATURE_PHYSICS_ENGINE_COLLISSION_FILTERING_ENABLED",
+                    "false"
```
Comment (robertpainsi) [inline build.gradle]: ...and this

**Comment by avihric** [comment]
retest this please

**Comment by avihric** [comment]
retest this please

**Comment by avihric** [comment]
retest this please

**Comment by sjaindl** [comment]
Finally got it working.. There have been a couple of issues.

First, the new gradle version does not support the method generateDebugTestBuildConfig anymore.. So these methods are no longer needed in build.gradle. Luckily, the BuildConfig can also be build using buildConfigField(..). This also allows to separate debug and release build types.

Second, the LookFragment was in certain situations detached from its Activity (ScriptActivity). Therefore, the activity reference is kept as member variable now when the fragment is attached.

Third, I also had to move the method calling the FragmentTransaction for switching to the ScriptFragment to the Activity.onCreate(). There was an issue where activity was already destroyed causing an Invalid-State-Exception and where potential state-loss may occur. This is pretty good explained at: http://www.androiddesignpatterns.com/2013/08/fragment-transaction-commit-state-loss.html

Fourth, I had to add an intent-filter in the Test-Manifest.. Otherwise an ActivityNotFound-Exception was thrown.. 

These changes are not only due to the gradle update, but also to other updates - especially, the Support Library I think.

New testrun (with 2 random fails): 
https://jenkins.catrob.at/job/Catroid-ParalellEmulatorUITest-CustomBranch/2478/

@avihric  @mhobl ready to merge ;)

**Comment by MarkusHobisch** [comment]
All tests works! 

https://jenkins.catrob.at/view/All-Categories/view/Catroid/job/Catroid-Device-Single-ClassAndPackage-Test/393/console
https://jenkins.catrob.at/view/All-Categories/view/Catroid/job/Catroid-Device-Single-ClassAndPackage-Test/394/console

**Comment by avihric** [comment]
@MarkusHobisch Actually both of your links are showing an aborted test run. I would not say that this is working :wonder, however @sjaindl posted already a successful run before.

**Comment by MarkusHobisch** [comment]
yes, but not the tests were failed. In the console output you will see that everythings works fine.

**Pull request merged**